compiler/rustc_codegen_llvm/src/builder.rs

```diff
@@ -937,8 +937,12 @@ impl BuilderMethods<'a, 'tcx> for Builder<'a, 'll, 'tcx> {
         pers_fn: &'ll Value,
         num_clauses: usize,
     ) -> &'ll Value {
+        // The LLVMBuildLandingPad requires personality to be a Function (as of LLVM 12), while we
+        // need a support for const bitcasts. Use LLVMSetPersonalityFn instead, which correctly
+        // casts personality to a Const.
```

Comment:
It is worth noting that setting personality_fn here this way is the right approach _either_ way. Its more direct and doesn't use legacy compatibility code.

Reply:
We could remove the call to `set_personality_fn`.

It does have the one advantage that it is impossible to forget to set a personality, and try intrinsics builds some landing pads manually.

Reply:
Yeah, if we're keeping it here where its impossible to forget then its probably worthwhile to remove calls such as [here](https://github.com/rust-lang/rust/blob/a55748ffe94e71f841c7b1d752779b0db138b342/compiler/rustc_codegen_ssa/src/mir/mod.rs#L151).

Reply:
I am not sure about this call in `codegen_mir`, it seems it would be insufficient for SEH implementation.

Generally, I would rather not modify here more things than necessary.